model/task_lifecycle.go

```diff
@@ -31,109 +31,114 @@ type StatusChanges struct {
 	BuildComplete    bool
 }
 
-func SetActiveState(t *task.Task, caller string, active bool) error {
-	originalTasks := []task.Task{*t}
-	if t.DisplayOnly {
-		execTasks, err := task.Find(task.ByIds(t.ExecutionTasks))
-		if err != nil {
-			return errors.Wrapf(err, "error getting execution tasks")
-		}
-		originalTasks = append(originalTasks, execTasks...)
-	}
-
-	if active {
-		// if the task is being activated and it doesn't override its dependencies
-		// activate the task's dependencies as well
-		tasksToActivate := []task.Task{}
-		if !t.OverrideDependencies {
-			deps, err := task.GetRecursiveDependenciesUp(originalTasks, nil)
+func SetActiveState(caller string, active bool, tasks ...*task.Task) error {
+	tasksToActivate := []task.Task{}
+	versionsToActivate := []Version{}
+	for _, t := range tasks {
+		originalTasks := []task.Task{*t}
+		originalTasks = append(originalTasks, *t)
+		if t.DisplayOnly {
+			execTasks, err := task.Find(task.ByIds(t.ExecutionTasks))
```

Comment:
Can we compile a list of all exec tasks and just query them once?

Reply:
I believe it would interfere with this block: https://github.com/evergreen-ci/evergreen/blob/main/model/task_lifecycle.go#L49

Where we are computing the dependencies of a single display task + all its execution tasks in one batch.  If we retrieved all of the relevant execution tasks for all tasks at once, I think we could be recomputing the same dependencies over and over again and adding them to the list.